ci-operator/step-registry/telco5g/origin/tests/telco5g-origin-tests-commands.sh

```diff
@@ -44,30 +41,236 @@ git clone -b "${CNF_BRANCH}" "${CNF_REPO}" cnf-features-deploy
+
+cd
+
+# Wait until number of nodes matches number of machines
+# Ref.: https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh
+for i in $(seq 30); do
+    nodes="$(oc get nodes --no-headers | wc -l)"
+    machines="$(oc get machines -A --no-headers | wc -l)"
+    [ "$machines" -le "$nodes" ] && break
+    sleep 30
+done
+
+[ "$machines" -le "$nodes" ]
+
+# Wait for nodes to be ready
+# Ref.: https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh
+oc wait nodes --all --for=condition=Ready=true --timeout=10m
+
+# Waiting for clusteroperators to finish progressing
+# Ref.: https://github.com/openshift/release/blob/master/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh
+oc wait clusteroperators --all --for=condition=Progressing=false --timeout=10m
+
+# Pull secrets might be required to extract url to container image of OpenShift's conformance test suite
+oc get secret pull-secret -n openshift-config -o jsonpath="{.data['\.dockerconfigjson']}" \
+    | base64 --decode > pull-secret.txt
+
+# Pull secrets are required to pull container image for openshift-tests
+oc get secret pull-secret -n openshift-config -o yaml | grep -v '^\s*namespace:\s' | oc apply -f -
+
+# Extract url to container image of OpenShift's conformance test suite aka openshift-tests
+T5_JOB_TESTS_IMAGE=$(oc adm release info --registry-config pull-secret.txt --image-for=tests "$T5_JOB_RELEASE_IMAGE")
+
+# Launch a pod from the openshift-tests image and let it idle
+# while we copy the openshift-tests binary out of the container
+oc run telco5g-tests-extractor --restart=Never --image "$T5_JOB_TESTS_IMAGE" --command=true \
+    --overrides='{ "spec": { "template": { "spec": { "imagePullSecrets": [{"name": "pull-secret"}] } } } }' \
+    -- sleep 3600
+
+oc wait pod/telco5g-tests-extractor --for=condition=Ready --timeout=15m
+
+# Extract openshift-tests binary from container
+# shellcheck disable=SC2034
+for i in $(seq 10); do # retry because copy is flaky
+    if oc cp telco5g-tests-extractor:/usr/bin/openshift-tests openshift-tests; then
+        break
+    else
+        rm -f openshift-tests # remove incompletely transferred files
+        sleep 5
+    fi
+done
+[ -e "openshift-tests" ]
+
+oc delete pods telco5g-tests-extractor
+
+chmod a+x openshift-tests
+
+# Determine list of tests
+./openshift-tests run --dry-run --provider "${TEST_PROVIDER}" "${TEST_SUITE}" > "$ARTIFACT_DIR/tests-all.txt"
+if [ -n "${TEST_SKIPS}" ]; then
+    grep -v "$TEST_SKIPS" "$ARTIFACT_DIR/tests-all.txt" > "$ARTIFACT_DIR/tests-run.txt"
+
+    if ! grep "$TEST_SKIPS" "$ARTIFACT_DIR/tests-all.txt" > "$ARTIFACT_DIR/tests-skip.txt"; then
+      echo >&2 "ERROR: No tests were found matching the TEST_SKIPS regex:"
+      echo >&2 "$TEST_SKIPS"
+      exit 1
+    fi
+
+    TEST_FILE=tests-run.txt
+else
+    TEST_FILE=tests-all.txt
+fi
+
```

Comment:
$TEST_FILE is worth to save to artifacts

Reply:
It is saved to `ARTIFACT_DIR`.